cpp/src/search/search.cu

```diff
@@ -129,4 +130,53 @@ gdf_column upper_bound(table const& t,
   return detail::search_ordered(t, values, false, desc_flags, nulls_as_largest);
 }
 
+template <bool nullable = true>
+struct compare_with_value{
+
+            compare_with_value(device_table t, device_table val, bool nulls_are_equal = true)
+                : compare(t, val, nulls_are_equal) {}
+
+            __device__ bool operator()(gdf_index_type i){
+               return compare(i, 0);
+            }
+            row_equality_comparator<nullable> compare;
+};
+
+bool contains(gdf_column const& column, gdf_scalar const& value)
+{
+  // No element to compare against
+  if (column.size == 0 || value.is_valid == false) {
```

Comment:
If `value.is_valid == false`, then you should return `cudf::has_nulls(column)`, i.e., if a user is searching for a null value, then return true if any null values exist in `column`.